main.js

```diff
@@ -17,6 +18,63 @@ function generateTiles() {
   return tiles
 }
 
+function isHidden(tile) {
+  return tile.originRow === 0 || tile.originRow === 7 || tile.originCol === 0 || tile.originCol === 7
+}
+
+function isHiddenCorner(tile) {
+  return (tile.originRow === 0 && tile.originCol === 0) || (tile.originRow === 0 && tile.originCol === 7) || (tile.originRow === 7 && tile.originCol === 7) || (tile.originRow === 7 && tile.originCol === 0)
+}
+
+function getGoalCandidates(board) {
+  let candidates = []
+  for (let i = 0; i < board.length; i++) {
+    for (let j = 0; j < board[i].length; j++) {
+      if (isHidden(board[i][j]) && !isHiddenCorner(board[i][j])) {
+        candidates.push(board[i][j])
+      }
+    }
+  }
+  return candidates
+}
+
+function shuffleArray(array) {
+  let shuffled = array.splice()
```

Comment:
Double-check [`splice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice?v=example) vs [`slice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice?v=example). Make sure you understand how these unfortunately-named functions work 😄 
Then you are approved to merge.